lib/id3c/cli/command/etl/manifest.py

```diff
@@ -95,55 +95,71 @@ def etl_manifest(*, db: DatabaseSession):
         with db.savepoint(f"manifest record {manifest_record.id}"):
             LOG.info(f"Processing record {manifest_record.id}")
 
-            # Convert sample barcode to full identifier, ensuring it's
-            # known and from the correct identifier set.
-            sample_barcode = manifest_record.document.pop("sample")
-            sample_identifier = find_identifier(db, sample_barcode)
+            # When updating an existing row, update the identifiers
+            # only if the record has both the 'sample' and
+            # 'collection' keys.
+            should_update_identifiers = "sample" in manifest_record.document \
+                and "collection" in manifest_record.document
 
-            if not sample_identifier:
-                LOG.warning(f"Skipping sample with unknown sample barcode «{sample_barcode}»")
-                mark_skipped(db, manifest_record.id)
-                continue
+             # Sample collection date
+            collected_date = manifest_record.document.pop("date", None)
 
-            if (manifest_record.document.get("sample_type") and
-                manifest_record.document["sample_type"] == "rdt"):
-                assert sample_identifier.set_name in expected_identifier_sets["rdt"], \
-                    (f"Sample identifier found in set «{sample_identifier.set_name}», " +
-                    f"not {expected_identifier_sets['rdt']}")
-            else:
-                assert sample_identifier.set_name in expected_identifier_sets["samples"], \
-                    (f"Sample identifier found in set «{sample_identifier.set_name}», " +
-                    f"not {expected_identifier_sets['samples']}")
-
-            # Optionally, convert the collection barcode to full
-            # identifier, ensuring it's known and from the correct
-            # identifier set.
+            # Attempt to find barcodes and their related identifiers
+            sample_barcode = manifest_record.document.pop("sample", None)
+            sample_identifier = find_identifier(db, sample_barcode) if sample_barcode else None
             collection_barcode = manifest_record.document.pop("collection", None)
             collection_identifier = find_identifier(db, collection_barcode) if collection_barcode else None
 
+            # Skip a record if it has no sample_identifier and no collection_identifier
+            if not sample_identifier and not collection_identifier:
+                LOG.warning(f"Skipping record «{manifest_record.id}» because it has no known sample barcode or collection barcode")
```

Comment:
Can we use the word "identifier" in this message instead of "barcode"?

Reply:
Also, can we change the language in this warning message to something slightly clearer like
> because it has neither a known sample identifier nor a collection identifier

?
The current wording suggests a sample could be missing one of the two identifiers and enter this block.

Reply:
I stole your sentence! Fixed.